### PR TITLE
fix #1791, work in progress

### LINKED ISSUE #1791: Use URL from ServerIdentity struct (JS only)
The task is to use the URL field in the ServerIdentity struct so that a roster can indicate how to contact the server from the client side.

Ideally, we'd do this work on all three client libraries (Go, Java, JavaScript). But to keep this task from being too big, let's deliver this feature on JS first, and then move on to other client libraries.

Tasks:
- [x] get dedis/student_18_explorer app working for dev mode, use the yarn version of `npm link` to work on a development version of cothority TS client.
- [x] fix cothority TS client to respect the URL field (see getWebSocketAddress). Hopefully callers will not notice the change at all (i.e. no corresponding change in the skipchain explorer?)
- [x] figure out how this can be tested in the context of the TypeScript tests -- perhaps it is enough to include one URL in the external/js/cothority/spec/support/public.toml file?
- [x] deploy it, and see it working for real
- [x] evolve one production conode to have URL in the ServerIdentity in it's roster (get help from @jeffallen )

### PR DESCRIPTION
**Pull request by nkcr**
This PR closes #1791

### CODE REVIEW

File: external/js/cothority/src/network/proto.ts [hunk 1]
```diff
@@ -178,6 +178,42 @@ export class ServerIdentity extends Message<ServerIdentity> {
...
+            case "http:": {
+                urlParser.protocol = "ws:";
+                break;
+            }
+            case "https:": {
+                urlParser.protocol = "wss:";
+                break;
+            }
+            default : {
+                throw new Error("The url field should use either 'http:' or 'https:', but We found "
```
Comment (jeffallen) [inline external/js/cothority/src/network/proto.ts]:
```suggestion
                throw new Error("The url field should use either 'http:' or 'https:', but we found "
```

File: external/proto/network.proto [hunk 2]
```diff
@@ -14,7 +14,9 @@ message ServerIdentity {
     required bytes id = 3;
     required string address = 4;
     required string description = 5;
-    optional string url = 6;
+	// Tag number 6 is used by go.dedis.ch/protobuf for the private
```
Comment (jeffallen) [inline external/proto/network.proto]:
```suggestion
    // Tag number 6 is used by go.dedis.ch/protobuf for the private
```

File: external/proto/network.proto [hunk 3]
```diff
@@ -14,7 +14,9 @@ message ServerIdentity {
...
+	// key field, which is not exported.
```
Comment (jeffallen) [inline external/proto/network.proto]:
```suggestion
    // key field, which is not exported.
```

**Comment by nkcr** [comment]
This PR closes #1791

File: conode/Operating.md [hunk 4]
```diff
@@ -11,19 +11,77 @@ line examples, please refer to:
...
+```
+
 ## Reverse proxy
 
 Conode should only be run as a non-root user.
 
 The current recommended way to add HTTPS to the websocket port is to use a web
 server like Apache or nginx in reverse proxy mode to forward connections from
-port 443 to port 6880, the default websocket port.
+port 443 the websocket port, which is the conode's port plus 1.
```
Comment (Gilthoniel) [inline conode/Operating.md]: `port 443 to the websocket port` ?

**Comment by kc1212** [comment]
Please update the PR description to say which issue this PR closes. The PR description goes into the commit logs but not the PR comments.

**Pull request merged**